cql3/restrictions/single_column_primary_key_restrictions.hh

```diff
@@ -355,22 +360,12 @@ public:
     virtual unsigned int num_prefix_columns_that_need_not_be_filtered() const override;
 };
 
+// FIXME: This code is dead, but we can't just remove it, because
+// single_column_primary_key_restrictions<clustering_key_prefix>::bounds_ranges() is not dead.
 template<>
 inline dht::partition_range_vector
 single_column_primary_key_restrictions<partition_key>::bounds_ranges(const query_options& options) const {
-    dht::partition_range_vector ranges;
-    ranges.reserve(size());
-    for (query::range<partition_key>& r : compute_bounds(options)) {
-        if (!r.is_singular()) {
-            throw exceptions::invalid_request_exception("Range queries on partition key values not supported.");
-        }
-        ranges.emplace_back(std::move(r).transform(
-            [this] (partition_key&& k) -> query::ring_position {
-                auto token = dht::get_token(*_schema, k);
-                return { std::move(token), std::move(k) };
-            }));
-    }
-    return ranges;
```

Comment:
Why is this dead code? because of the expr conversions? -> changelog.

Reply:
I don't know for sure when this code died, but it seems that it happened about two years ago in 4bd0cb8dd9309237c8e107e86432c09cd2fec9ca and related patches.

There are `// FIXME(sarna): transitive hack only, do not judge. Should be dropped after all primary_key_restrictions<T> uses are removed from code.` and `// NOTICE(sarna): This function is useless for partition key restrictions, but it should remain here until single_column_primary_key_restrictions class is detemplatized.` scattered around. It seems that some refactoring was unfinished and some dead code remained.

I only noticed it because `token_restriction::bounds_ranges` contains some token-related code, which I carefully rewrote only to realize that it's not called anywhere.

Reply:
@psarna Do you remember this? What are those leftovers?

Reply:
I don't remember exactly, but it's entirely possible that this code died when I partially detemplatized the restrictions. The effort was later dropped because it was decided that this whole layer is going to be rewritten with expressions anyway.